libs/platform/core/src/services/entity/entity-provider.ts

```diff
@@ -0,0 +1,34 @@
+import { Provider } from '@spryker-oryx/di';
+
+export const EntityProvider = 'oryx.EntityProvider*';
+
+export interface EntityProvider<E = unknown, Q = unknown> {
+  service: string;
+  // factory?: () => E;
+  context?: string;
+  // contextToQualifier?: () => Q;
+}
+
+export function provideEntity(type: string, config: EntityProvider): Provider;
+export function provideEntity(
+  type: string,
+  factory: () => EntityProvider
+): Provider;
```

Comment:
do we need this overloading? I'd think leaving it would works perfectly fine for ts